joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -929,6 +929,28 @@ class AstCreator(filename: String, global: Global) {
     AstWithCtx(switchAst, ctx)
   }
 
+  private def astForSynchronizedStatement(stmt: SynchronizedStmt,
+                                          scopeContext: ScopeContext,
+                                          order: Int): AstWithCtx = {
+    val parentNode =
+      NewBlock()
```

Comment:
wouldn't it be better for this to be a modifier? https://cpg.joern.io/#node-ref-modifier